lndclient/lightning_client.go

```diff
@@ -109,6 +112,132 @@ type ChannelInfo struct {
 	Uptime time.Duration
 }
 
+// ClosedChannel represents a channel that has been closed.
+type ClosedChannel struct {
+	// ChannelPoint is the funding outpoint of the channel.
+	ChannelPoint string
+
+	// ChannelID holds the unique channel ID for the channel. The first 3
+	// bytes are the block height, the next 3 the index within the block,
+	// and the last 2 bytes are the /output index for the channel.
+	ChannelID uint64
+
+	// ClosingTxHash is the tx hash of the close transaction for the channel.
+	ClosingTxHash string
+
+	// CloseType is the type of channel closure.
+	CloseType CloseType
+
+	// CloseInitiator indicates which party initiated the channel close.
+	// Since this value is not always set in the rpc response, we also make
+	// a best effort attempt to set it based on CloseType.
+	CloseInitiator CloseInitiator
+
+	// PubKeyBytes is the raw bytes of the public key of the remote node.
+	PubKeyBytes route.Vertex
+
+	// Capacity is the total amount of funds held in this channel.
+	Capacity btcutil.Amount
+
+	// SettledBalance is the amount we were paid out directly in this
+	// channel close. Note that this does not include cases where we need to
+	// sweep our commitment or htlcs.
+	SettledBalance btcutil.Amount
+}
+
+// CloseType is an enum which represents the types of closes our channels may
+// have. This type maps to the rpc value.
+type CloseType uint8
+
+const (
+	// CloseTypeCooperative represents cooperative closes.
+	CloseTypeCooperative CloseType = iota
+
+	// CloseTypeLocalForce represents force closes that we initiated.
+	CloseTypeLocalForce
+
+	// CloseTypeRemoteForce represents force closes that our peer initiated.
+	CloseTypeRemoteForce
+
+	// CloseTypeBreach represents breach closes from our peer.
+	CloseTypeBreach
+
+	// CloseTypeFundingCancelled represents channels which were never
+	// created because their funding transaction was cancelled.
+	CloseTypeFundingCancelled
+
+	// CloseTypeAbandoned represents a channel that was abandoned.
+	CloseTypeAbandoned
+)
+
+func (c CloseType) String() string {
+	switch c {
+	case CloseTypeCooperative:
+		return "Cooperative"
+
+	case CloseTypeLocalForce:
+		return "Local Force"
```

Comment:
I'm using them in Faraday for user reports. Reasoning for not using snake/camel was that if we go through the bother of making this new enum, maybe it should be the user-friendly version? Otherwise we really might as well just use the rpc generated enums. 

I'm not sure how decided I am on keeping lndclient strictly non-rpc types, it's a lot of boilerplate to copy all of these across.

Reply:
Okay, if you use them in the reports it's nice to have them in human readable/natural form.